moveit_simple/src/moveit_simple.cpp

```diff
@@ -143,7 +187,82 @@ void Robot::addTrajPoint(const std::string & traj_name, const std::string & poin
 }
 
 
-  
+
+void Robot::addTrajPoint(const std::string & traj_name, const std::string & point_name,
+                         const std::string & custom_tool_frame, 
+                         double time, const InterpolationType & type,
+                         const unsigned int num_steps)
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+
+  ROS_INFO_STREAM("Attempting to add " << point_name << " to " << traj_name <<
+                 " at time " << time <<
+                 " and custom_tool_frame [" << custom_tool_frame << "]");
+
+  // Check to see if the point under consideration is a Joint point/state
+  if ( virtual_robot_state_->setToDefaultValues(joint_group_, point_name) )
+  {
+    ROS_INFO_STREAM("Attempting to add joint point/state [" << point_name << "] to trajectory");
+
+    try
+    {
+      std::unique_ptr<TrajectoryPoint> point = lookupTrajectoryPoint(point_name, time);
+      addTrajPoint(traj_name, point, type, num_steps);
+    }
+    catch ( std::invalid_argument &ia )
+    {
+      ROS_ERROR_STREAM("Invalid point " << point_name << " to add to " << traj_name);
+      throw ia;
+    }
+    catch ( tf2::TransformException &ex )
+    {
+      ROS_ERROR_STREAM(" TF transform failed for " << point_name << " to add to " << traj_name);
+      throw ex;
+    } 
+  }
+
+  else  
+  {
+    try
+    {
+      ROS_INFO_STREAM("Looked up cartesian target [" << point_name << 
+                      "] from robot model: ");
+
+      // We take the Origin of our reference frame in consideration as the "known pose"
+      // In this case our reference frame resolves to the variable "point_name"
+      Eigen::Affine3d pose_eigen_buffer;
+      geometry_msgs::Pose pose_buffer;
+
+      pose_buffer.position.x = 0.000;
```

Comment:
Don't change anything, but the default constructors should make initialize these values.  There isn't a need to do this (but it doesn't hurt, so leave it in).